plugins/org.eclipse.n4js.transpiler/src/org/eclipse/n4js/transpiler/TransformationDependency.java

```diff
@@ -101,6 +129,34 @@
 	}
 
 	/**
+	 * Tells if the given transformation is active, based on the given {@link GeneratorOption generator options}.
+	 */
+	public static final boolean isActiveIn(Transformation transformation, GeneratorOption[] activeOptions) {
+		final Optional ann = transformation.getClass().getAnnotation(Optional.class);
+		if (ann != null) {
+			// at least one of the options given in the annotation must be a super type of any of the active options
```

Comment:
what do you mean with `super type`? We are looking up the declared annotations here and not testing `instanceof` anywhere.